src/presentation/http/router/note.ts

```diff
@@ -111,14 +118,24 @@ const NoteRouter: FastifyPluginCallback<NoteRouterOptions> = (fastify, opts, don
     ],
   }, async (request, reply) => {
     const { note } = request;
+    const noteId = request.note?.id as number;
     const { memberRole } = request;
+    const { userId } = request;
 
     /**
      * Check if note exists
      */
     if (note === null) {
       return reply.notFound('Note not found');
     }
+
+    /**
+     * Check if user is authorized
+     */
+    if (userId !== null) {
+      await noteVisitsService.saveVisit(noteId, userId);
```

Comment:
I think, it is a part of the business logic, so it would be better to move it on domain level